### PR TITLE
Add Blobstore Servlets

### PR DESCRIPTION
**Pull request by TadjAdams**

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/Serve.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+@WebServlet("/Serve")
```
Comment (Gateswong) [inline src/main/java/com/google/sps/servlets/Serve.java:29]: _Formatting Advice_

Put a new line before this.

File: src/main/java/com/google/sps/servlets/Serve.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class Serve extends HttpServlet {
+    private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+ 
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res)
+        throws IOException {
```
Comment (Gateswong) [inline src/main/java/com/google/sps/servlets/Serve.java:35]: Is this intended? Everytime you are accessing /Serve you will get 502 error from server.

File: src/main/java/com/google/sps/servlets/Serve.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
```
Comment (jojojames) [inline src/main/java/com/google/sps/servlets/Serve.java:28]: Need line between

File: src/main/java/com/google/sps/servlets/Serve.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+@WebServlet("/Serve")
+public class Serve extends HttpServlet {
+    private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+ 
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res)
+        throws IOException {
+            BlobKey blobKey = new BlobKey(req.getParameter("blob-key"));
+            blobstoreService.serve(blobKey, res);
+        }
```
Comment (jojojames) [inline src/main/java/com/google/sps/servlets/Serve.java:38]: Bracket wrong place, match with public

File: src/main/java/com/google/sps/servlets/Serve.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+@WebServlet("/Serve")
+public class Serve extends HttpServlet {
+    private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+ 
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res)
+        throws IOException {
+            BlobKey blobKey = new BlobKey(req.getParameter("blob-key"));
```
Comment (jojojames) [inline src/main/java/com/google/sps/servlets/Serve.java:36]: Indented by 4

File: src/main/java/com/google/sps/servlets/Upload.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+ 
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res)
+        throws ServletException, IOException {
+ 
+        Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(req);
+        List<BlobKey> blobKeys = blobs.get("myFile");
+ 
+        if (blobKeys == null || blobKeys.isEmpty()) {
```
Comment (jojojames) [inline src/main/java/com/google/sps/servlets/Upload.java:41]: Can use a comment on what's happening

File: src/main/java/com/google/sps/servlets/Upload.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+@WebServlet("/Upload")
+public class Upload extends HttpServlet {
+    private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+ 
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res)
+        throws ServletException, IOException {
+ 
+        Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(req);
+        List<BlobKey> blobKeys = blobs.get("myFile");
```
Comment (jojojames) [inline src/main/java/com/google/sps/servlets/Upload.java:39]: These "myFile" can be constants

File: src/main/webapp/script.js [hunk 8]
```diff
@@ -61,4 +61,18 @@ function drawChart() {
         document.getElementById('chart-container'));
     chart.draw(data, options);
   });
-}
\ No newline at end of file
+}
+
+//Fetches image data
+
```
Comment (jojojames) [inline src/main/webapp/script.js:67]: Random extra line

File: src/main/webapp/script.js [hunk 9]
```diff
@@ -61,4 +61,18 @@ function drawChart() {
         document.getElementById('chart-container'));
     chart.draw(data, options);
   });
-}
\ No newline at end of file
+}
+
+//Fetches image data
```
Comment (jojojames) [inline src/main/webapp/script.js:66]: Space after //

File: src/main/webapp/script.js [hunk 10]
```diff
@@ -61,4 +61,18 @@ function drawChart() {
...
+
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/Serve')
```
Comment (jojojames) [inline src/main/webapp/script.js:69]: Add comments on what is being done

**Comment by jojojames** [comment]
This shouldn’t be merged until review comments are addressed and then another review takes place with an LGTM.

**Pull request merged**